pkg/git/factory/gitfactory.go

```diff
@@ -36,13 +46,23 @@ func Build(ctx context.Context, cluster *v1alpha1.Cluster, fluxConfig *v1alpha1.
 			return nil, err
 		}
 
+		tools.Provider, err = buildGithubProvider(ctx, githubToken, fluxConfig.Spec.Github)
+		if err != nil {
+			return nil, fmt.Errorf("building github provider: %v", err)
+		}
+
+		gitAuth = &http.BasicAuth{Password: githubToken, Username: fluxConfig.Spec.Github.Owner}
 		repo = fluxConfig.Spec.Github.Repository
 		repoUrl = github.RepoUrl(fluxConfig.Spec.Github.Owner, repo)
-		tokenAuth = &git.TokenAuth{Token: githubToken, Username: fluxConfig.Spec.Github.Owner}
-		tools.Provider, err = buildGithubProvider(ctx, *tokenAuth, fluxConfig.Spec.Github)
+	case fluxConfig.Spec.Git != nil:
```

Comment:
I don't like the idea of having a case here like this, all of this information should ideally be abstracted into methods for a specific provider.

Reply:
see above comment and https://github.com/aws/eks-anywhere/issues/1993